codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpData.java

```diff
@@ -99,6 +99,23 @@
     long length();
 
     /**
+     * Returns the current defined length of the HttpData.
+     *
+     * If no Content-Length is provided in the request, this is equivalent to the current size (length()) since the
+     * decoding does not known the final size yet. Therefore the defined length is
+     * always 0 the current length (whatever during decoding or in final state).
+     *
+     * If Content-Length is provided in the request, this is this given defined length.
+     * This value does not changed, whatever during decoding or in the final state.
```

Comment:
During decoding, the FileUpload (for instance, but the same for an Attribute) is growing in size (chunk by chunk, through `length()` method), therefore its current size is growing, while the `defined size` is not changing. Once fully decoded, the `lengh()` and `definedLength()` shall return the same size.